diff --git a/app/osd/horizonladder.cpp b/app/osd/horizonladder.cpp
index f0d3516..c1654b2 100644
--- a/app/osd/horizonladder.cpp
+++ b/app/osd/horizonladder.cpp
@@ -53,8 +53,14 @@ void HorizonLadder::paint(QPainter* painter) {
     }
     //weird rounding issue where decimals make ladder dissappear
     roll_degree = round(roll_degree);
-    pitch_degree = round(pitch_degree);
-    //qDebug()<<"Roll:"<<roll_degree<<" Pitch:"<<pitch_degree;
+    //pitch_degree = round(pitch_degree);
+
+    std::string str = std::to_string(pitch_degree);
+    qDebug()<<"Roll:"<<roll_degree<<" Pitch:"<<pitch_degree<< " : " << str.c_str();
+
+    
+
+
 
     if(show_center_indicator){
         painter->setPen(m_color);
@@ -446,7 +452,7 @@ void HorizonLadder::setRoll(int roll) {
 }
 
 
-void HorizonLadder::setPitch(int pitch) {
+void HorizonLadder::setPitch(double pitch) {
     m_pitch = pitch;
     emit pitchChanged(m_pitch);
     update();
diff --git a/app/osd/horizonladder.h b/app/osd/horizonladder.h
index d113b7e..af2df33 100644
--- a/app/osd/horizonladder.h
+++ b/app/osd/horizonladder.h
@@ -18,7 +18,7 @@ class HorizonLadder : public QQuickPaintedItem {
     Q_PROPERTY(int horizonRange MEMBER m_horizonRange WRITE setHorizonRange NOTIFY horizonRangeChanged)
     Q_PROPERTY(int horizonStep MEMBER m_horizonStep WRITE setHorizonStep NOTIFY horizonStepChanged)
     Q_PROPERTY(int roll MEMBER m_roll WRITE setRoll NOTIFY rollChanged)
-    Q_PROPERTY(int pitch MEMBER m_pitch WRITE setPitch NOTIFY pitchChanged)
+    Q_PROPERTY(double pitch MEMBER m_pitch WRITE setPitch NOTIFY pitchChanged)
 
     Q_PROPERTY(int heading MEMBER m_heading WRITE setHeading NOTIFY headingChanged)
     Q_PROPERTY(int homeHeading MEMBER m_homeHeading WRITE setHomeHeading NOTIFY homeHeadingChanged)
@@ -52,7 +52,7 @@ public slots:
     void setHorizonRange(int horizonRange);
     void setHorizonStep(int horizonStep);
     void setRoll(int roll);
-    void setPitch(int pitch);
+    void setPitch(double pitch);
 
     void setHeading(int heading);
     void setHomeHeading(int homeHeading);
@@ -95,7 +95,7 @@ private:
     int m_horizonRange;
 
     int m_roll;
-    int m_pitch;
+    double m_pitch;
 
     int m_heading;
     int m_homeHeading;
